### PR TITLE
CXX: Improve parsing of function-try-blocks and try/catch in general

### PR DESCRIPTION
**Pull request by pragmaware**
Improved solution for #1746.

-  catch() is part of the function-try-block:
-- the variables declared inside now have proper scope
-- the function ends after the last catch()
- variables declared in the catch() parenthesis are extracted
- function-try-blocks are reported as properties

### CODE REVIEW

File: parsers/cxx/cxx_parser.c [hunk 1]
```diff
@@ -1617,76 +1622,114 @@ bool cxxParserParseAccessSpecifier(void)
...
-			cxxTokenChainDestroyFirst(pChain);
-			// Fake the final semicolon
-			CXXToken * t = cxxTokenChainLast(pChain);
-			t->eType = CXXTokenTypeSemicolon;
-			vStringClear(t->pszWord);
-			vStringPut(t->pszWord,';');
-
-			// and extract variable declarations if possible
-			cxxParserExtractVariableDeclarations(pChain,0);
+					);
```
Comment (masatake) [inline parsers/cxx/cxx_parser.c]: Visual Studio makes a warning on this line (https://ci.appveyor.com/project/universalctags/ctags/build/1.0.763/job/wim07v3kr02a88r6). How about using trigraph like:
```
bOkToExtractVariables = cxxTokenChainNextTokenOfType(
						pAndOrStar,
 						CXXTokenTypeAssignment | CXXTokenTypeSemicolon
					)? true: false;
```

or ...
```
} else if ( cxxTokenChainNextTokenOfType(
					pAndOrStar,
 						CXXTokenTypeAssignment | CXXTokenTypeSemicolon)) {
            bOkToExtractVariables = true;
}

File: parsers/cxx/cxx_tag.c [hunk 2]
```diff
@@ -286,6 +286,9 @@ vString * cxxTagSetProperties(unsigned int uProperties)
 		vStringCatS(pszProperties,_szProperty); \
 	} while(0)
 
+	// FIXME: The property names might be too verbose. Maybe it could be a good idea
+	//        to a switch to shorten them to 1-2 letter codes?
+
```
Comment (masatake) [inline parsers/cxx/cxx_tag.c]: I think basically using long names is not bad.
Making shortere can be done in a post processor.

File: Units/parser-cxx.r/function_try_block.d/args.ctags [hunk 3]
```diff
@@ -1 +1,4 @@
+--c++-kinds=+pflz
```
[the remark this answers is not in the document]
Comment (pragmaware) [reply Units/parser-cxx.r/function_try_block.d/args.ctags]: done

File: parsers/cxx/cxx_parser.c (hunk 1 above)
Comment (pragmaware) [reply parsers/cxx/cxx_parser.c]: done

File: parsers/cxx/cxx_tag.c (hunk 2 above)
Comment (pragmaware) [reply parsers/cxx/cxx_tag.c]: OK

**Comment by masatake** [comment]
Thank you.

**Pull request merged**